src/bin/stellar-xdr/guess.rs

```diff
@@ -0,0 +1,227 @@
+use std::{
+    cmp,
+    fs::File,
+    io::{self, stdin, Read},
+    path::PathBuf,
+};
+
+use clap::{Args, ValueEnum};
+
+use crate::Channel;
+
+#[derive(thiserror::Error, Debug)]
+#[allow(clippy::enum_variant_names)]
+pub enum Error {
+    #[error("error decoding XDR: {0}")]
+    ReadXdrCurr(#[from] stellar_xdr::curr::Error),
+    #[error("error decoding XDR: {0}")]
+    ReadXdrNext(#[from] stellar_xdr::next::Error),
+    #[error("error reading file: {0}")]
+    ReadFile(#[from] std::io::Error),
+}
+
+#[derive(Args, Debug, Clone)]
+#[command()]
+pub struct Cmd {
+    /// File to decode, or stdin if omitted
+    #[arg()]
+    file: Option<PathBuf>,
+
+    // Input format of the XDR
+    #[arg(long, value_enum, default_value_t)]
+    input: InputFormat,
+
+    // Output format
+    #[arg(long, value_enum, default_value_t)]
+    output: OutputFormat,
+
+    /// Certainty as an arbitrary value
```

Comment:
Is this the number of suggestions to process?

Reply:
It's the number of value reads to make on a stream, if it is a stream, to confirm they continue to decode fine before accepting that guess. I didn't write this into the comment because I wanted to keep it somewhat opaque beyond the general idea that bigger number == more confidence.